sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_utilities.hpp

```diff
@@ -12,8 +12,16 @@
 
 namespace Azure { namespace Storage { namespace Files { namespace DataLake { namespace _detail {
 
-  std::string GetBlobUrlFromUrl(const std::string& url);
-  std::string GetDfsUrlFromUrl(const std::string& url);
+  Azure::Core::Url GetBlobUrlFromUrl(const Azure::Core::Url& url);
+  Azure::Core::Url GetDfsUrlFromUrl(const Azure::Core::Url& url);
+  inline std::string GetBlobUrlFromUrl(const std::string& url)
+  {
+    return GetBlobUrlFromUrl(Azure::Core::Url(url)).GetAbsoluteUrl();
+  }
+  inline std::string GetDfsUrlFromUrl(const std::string& url)
+  {
+    return GetDfsUrlFromUrl(Azure::Core::Url(url)).GetAbsoluteUrl();
+  }
```

Comment:
I don't understand what these are doing, especially when the names of `Url` accepting ones are identical to the one accepting `std::string`. Add some comments.

Reply:
It's overloaded functions intended to transform between dfs endpoints and blob endpoints.

Like `someaccount.dfs.core.windows.net` <-> `someaccount.blob.core.windows.net`.

They're in detail namespace.